### PR TITLE
fix(lsp): Ensure users get feedback on references/symbols errors or empty results

### PR DESCRIPTION
**Pull request by mfussenegger**
Relates to https://github.com/neovim/neovim/issues/15050

Users should get some indication if there was an error or an empty
result.

### CODE REVIEW

File: runtime/lua/vim/lsp/handlers.lua [hunk 1]
```diff
@@ -191,30 +191,37 @@ M['textDocument/codeLens'] = function(...)
...
 
-  util.set_qflist(util.symbols_to_items(result, bufnr))
-  api.nvim_command("copen")
+--@private
+--- Return a function that converts LSP responses to quickfix items and opens the qflist
+--
+--@param map_result function `((resp, bufnr) -> list)` to convert the response
+--@param entity name of the resource used in a `not found` error message
+local function response_to_qflist(map_result, entity)
+  return function(err, _, result)
```
Comment (mjlbach) [inline runtime/lua/vim/lsp/handlers.lua:202]: bufnr?

Comment (mfussenegger) [reply runtime/lua/vim/lsp/handlers.lua]: Right 🤦
Fixed it.

**Review by mjlbach — APPROVED** [review]

**Comment by mfussenegger** [comment]
I just was that the error handling is also inconsistent:

```
  if err then
    error("Could not execute code action: "..err.message)
  end
```

```
  if err then error(vim.inspect(err)) end
```


```
    if err then
      return err_message(tostring(err))
    end
```


Should we normalize that as well?

**Comment by mjlbach** [comment]
Up to you, we should probably use the err_message utility function instead of vim.inspect for consistency, I assume the string conversion was added initially in response to a specific issue, but that predates me.

**Comment by mfussenegger** [comment]
I'm a bit too trigger happy today.
Turns out the error handling within the handlers didn't work to begin with because of this:

https://github.com/neovim/neovim/blob/256570a7a628aa377f405cb55343a38d186dc770/runtime/lua/vim/lsp/handlers.lua#L456-L458

And that results on a `table: 0x.......` message - not very helpful.
I'll add a second commit that fixes it.

**Review by mjlbach — APPROVED** [review]

**Pull request merged**